query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/QueryRequestToPinotSQLConverter.java

```diff
@@ -164,16 +171,65 @@ private String convertFilter2String(Filter filter, Params.Builder paramsBuilder)
           builder.append(convertLiteralToString(kvp[MAP_VALUE_INDEX], paramsBuilder));
           break;
         default:
+          rhs = handleValueConversionForLiteralExpression(filter.getLhs(), filter.getRhs());
           builder.append(convertExpression2String(filter.getLhs(), paramsBuilder));
           builder.append(" ");
           builder.append(operator);
           builder.append(" ");
-          builder.append(convertExpression2String(filter.getRhs(), paramsBuilder));
+          builder.append(convertExpression2String(rhs, paramsBuilder));
       }
     }
     return builder.toString();
   }
 
+  /**
+   * Handles value conversion of a literal expression based on its associated column.
+   *
+   * @param lhs LHS expression with which literal is associated with
+   * @param rhs RHS expression which needs value conversion if its a literal expression
+   * @return newly created literal {@link Expression} of rhs if converted else the same one.
+   */
+  private Expression handleValueConversionForLiteralExpression(Expression lhs, Expression rhs) {
+    if (!(lhs.getValueCase().equals(COLUMNIDENTIFIER) && rhs.getValueCase().equals(LITERAL))) {
+      return rhs;
+    }
+
+    Expression newRhs = rhs;
+    String lhsColumnName = lhs.getColumnIdentifier().getColumnName();
+
+    if (viewDefinition.isBytesColumn(lhsColumnName)) {
```

Comment:
I think we should rather pass the column type to the `ToValueConverter` and let the implementation take care what to do for string vs bytes columns. We can avoid this if check and special logic for Bytes with that.

the converter instance will take care of bytes and other cases so why do we need to check for `isBytesColumn` explicitly?

Reply:
Sounds good. I will make function generic one.

Reply:
Sounds good, will make it generic. I was avoiding StringToString conversion as default types is a column and re-creation of the Value object.

Done.